server/parsec/components/memory/datamodel.py

```diff
@@ -68,6 +70,8 @@ class MemoryOrganization:
     vlobs: dict[VlobID, list[MemoryVlobAtom]] = field(default_factory=dict)
     blocks: dict[BlockID, MemoryBlock] = field(default_factory=dict)
     block_store: dict[BlockID, bytes] = field(default_factory=dict, repr=False)
+    # The user id is the author of the shamir recovery process
+    shamir_setup: dict[UserID, MemoryShamirSetup] = field(default_factory=dict)
```

Comment:
Contrary to v2, shamir certificates need to be kept even when they're outdated by a new setup, due to those certificates being part of the shamir topic.

It's probably enough to turn `shamir_setup` into a `dict[UserID, List[MemoryShamirSetup]]` for the moment, although the postgresql tables will probably be trickier than that.

Also, this doesn't take shamir removal into account, so this attribute might eventually become a `dict[UserID, List[MemoryShamirSetup | MemoryShamirRemoval]]`